### PR TITLE
Add MCP2221 support.

### PR DESCRIPTION
**Pull request by caternuson**
This has initial working MCP2221 code.

### CODE REVIEW

File: src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+    def i2c_readfrom_into(self, address, buffer, *, start=0, end=None):
+        end = end if end else len(buffer)
+        retries = 0
+        while retries < 5:
+            #
+            # why does this require two xfers?
+            #
+            # 1. tell it we want to read
+            self._hid_xfer(bytes([0x91,                     # i2c read data
```
Comment (ladyada) [inline src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py]: i had to use command 0x93 here

Comment (caternuson) [reply src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py]: what sensor?

Comment (ladyada) [reply src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py]: it was msa301 i think?

File: src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    GP_GPIO = 0b000
+    GP_DEDICATED = 0b001
+    GP_ALT0 = 0b010
+    GP_ALT1 = 0b011
+    GP_ALT2 = 0b100
+
+    def __init__(self):
+        self._hid = hid.device()
+        self._hid.open(MCP2221.VID, MCP2221.PID)
+        self._reset()
```
[the remark this answers is not in the document]
Comment (caternuson) [reply src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py]: there, or should it be part of `_reset()`?

[the remark this answers is not in the document]
Comment (ladyada) [reply src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py]: post - altho, not sure if it really fixed the upsettnesss :) see my big PR for all changes!

**Pull request merged**